rust/lang/security/audit/crypto/deprecated-ssl.yaml

```diff
@@ -0,0 +1,50 @@
+rules:
+  - id: ssl-v3-is-insecure
+    message: |
+      SSLv3 is insecure because it has known vulnerabilities.
+      Instead, use 'SslVersion::TLS1_2 or SslVersion::TLS1_3'.
+    metadata:
+      cwe: 'CWE-327: Use of a Broken or Risky Cryptographic Algorithm'
+      owasp: 'A9: Using Components with Known Vulnerabilities'
+      references:
+        - https://golang.org/doc/go1.14#crypto/tls
+        - https://www.us-cert.gov/ncas/alerts/TA14-290A
+    languages: [rust]
+    severity: ERROR
+    fix-regex:
+      regex: SslVersion::SSL3
+      replacement: SslVersion::TLS1_3
```

Comment:
If there are fixes, it is possible to add fix tests.

Create a new file called `deprecated-ssl.fixed.rs` with the expected outcome of running the autofix on the `deprecated-ssl.rs` file.

More info here: https://semgrep.dev/docs/writing-rules/testing-rules/#testing-autofix

`fix-regex` works but using regular `fix` will be more powerful, especially when we develop AST-based autofix for Rust. (https://semgrep.dev/blog/2022/autofixing-code-with-semgrep/)

If you use `fix` your rules will likely be more future-proof. Your rule would like something like this:

```
patterns:
  - pattern: |
        $VAR.set_min_proto_version(Some($TLS))
  - metavariable-regex:
        metavariable: $TLS
        regex: SslVersion::(SSL3|TLSv1|TLSv1_1)
  - focus-metavariable: $TLS
fix: |
  SslVersion::TLS1_3
```

This would allow you to combine all three rules into just one rule! You can even use the metavariable in the message if you want to still keep three separate messages:

```
message: >-
    $TLS is insecure ...
```